homeassistant/components/bosch_alarm/config_flow.py

```diff
@@ -129,6 +143,65 @@ async def async_step_user(
             errors=errors,
         )
 
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle DHCP discovery."""
+        self.mac = format_mac(discovery_info.macaddress)
+        self.host = discovery_info.ip
+        if self.hass.config_entries.flow.async_has_matching_flow(self):
+            return self.async_abort(reason="already_in_progress")
+
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if entry.data[CONF_MAC] == self.mac:
+                result = self.hass.config_entries.async_update_entry(
+                    entry,
+                    data={
+                        **entry.data,
+                        CONF_HOST: discovery_info.ip,
+                    },
+                )
+                if result:
+                    self.hass.config_entries.async_schedule_reload(entry.entry_id)
+                return self.async_abort(reason="already_configured")
+            if entry.data[CONF_HOST] == discovery_info.ip:
+                result = self.hass.config_entries.async_update_entry(
+                    entry,
+                    data={
+                        **entry.data,
+                        CONF_MAC: self.mac,
+                    },
+                )
+                if result:
+                    self.hass.config_entries.async_schedule_reload(entry.entry_id)
+                return self.async_abort(reason="already_configured")
```

Comment:
why would the mac change? If we find a new mac, the user should create a new entry

Reply:
good call, ill get rid of that